src/Sulu/Bundle/ContactBundle/Content/Types/ContactSelectionContentType.php

```diff
@@ -0,0 +1,212 @@
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var IdConverterInterface
+     */
+    private $converter;
+
+    /**
+     * @var IndexComparatorInterface
+     */
+    private $comparator;
+
+    public function __construct(
+        $template,
+        ContactManagerInterface $contactManager,
+        ContactManagerInterface $accountManager,
+        SerializerInterface $serializer,
+        IdConverterInterface $converter,
+        IndexComparatorInterface $comparator
+    ) {
+        $this->template = $template;
+        $this->contactManager = $contactManager;
+        $this->accountManager = $accountManager;
+        $this->serializer = $serializer;
+        $this->converter = $converter;
+        $this->comparator = $comparator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType()
+    {
+        return ContentTypeInterface::PRE_SAVE;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read(
+        NodeInterface $node,
+        PropertyInterface $property,
+        $webspaceKey,
+        $languageCode,
+        $segmentKey
+    ) {
+        $values = [];
+        if ($node->hasProperty($property->getName())) {
+            $values = $node->getPropertyValue($property->getName());
+        }
+        $this->setData($values, $property);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readForPreview(
+        $data,
+        PropertyInterface $property,
+        $webspaceKey,
+        $languageCode,
+        $segmentKey
+    ) {
+        $this->setData($data, $property);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(
+        NodeInterface $node,
+        PropertyInterface $property,
+        $userId,
+        $webspaceKey,
+        $languageCode,
+        $segmentKey
+    ) {
+        $value = $property->getValue();
+        $node->setProperty($property->getName(), ($value === null ? [] : $value));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove(
+        NodeInterface $node,
+        PropertyInterface $property,
+        $webspaceKey,
+        $languageCode,
+        $segmentKey
+    ) {
+        if ($node->hasProperty($property->getName())) {
+            $node->getProperty($property->getName())->remove();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContentData(PropertyInterface $property)
+    {
+        $value = $property->getValue();
+        $locale = $property->getStructure()->getLanguageCode();
+
+        if ($value === null || !is_array($value) || count($value) === 0) {
+            return [];
+        }
+
+        $ids = $this->converter->convertIdsToGroupedIds($value, ['a' => [], 'c' => []]);
+
+        $accounts = $this->accountManager->getByIds($ids['a'], $locale);
+        $contacts = $this->contactManager->getByIds($ids['c'], $locale);
+
+        $result = array_merge($accounts, $contacts);
+        @usort(
```

Comment:
is `@` is added because of a php bug ... https://bugs.php.net/bug.php?id=50688 and http://stackoverflow.com/questions/3235387/usort-array-was-modified-by-the-user-comparison-function